### ISSUE TITLE
LintDiff: Need equivalent of `PatchBodyParametersSchema` rule so we can disable

### ISSUE THREAD

**Issue by tjprescott** [Azure/typespec-azure#406]
https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-body-parameters-schema.md

**Comment by tjprescott** [comment]
Here's the issue that is needed to unblock this: https://github.com/microsoft/typespec/issues/3012

**Comment by tjprescott** [comment]
This is also blocked because it would require ARM approval since it applies to DataPlane *and* ARM specs.

**Comment by markcowl** [comment]
Closing in favor of #4290
